src/api/VoteMonitor.Api.DataExport/Handlers/DataExportQueryHandler.cs

```diff
@@ -76,12 +75,53 @@ public async Task<List<ExportModel>> Handle(GetDataForExport request, Cancellati
 			LEFT JOIN Notes n
 				ON n.IdQuestion = q.Id AND n.IdObserver = obs.Id AND n.IdPollingStation = a.IdPollingStation
 		WHERE
-			a.LastModified >= @LastModified
+			a.LastModified >= @from
 			AND a.IdObserver > @IdObserverLimit
-			AND obs.IdNgo <> @IdNgo
-            ", new SqlParameter("LastModified", new DateTime(2019, 11, 08, 6, 0, 0))
-                , new SqlParameter("IdObserverLimit", 10)
-                , new SqlParameter("IdNgo", 1));
+            
+            ";
+
+            var parameters = new List<SqlParameter>
+            {
+                new SqlParameter("@from", request.From ?? new DateTime(2019, 11, 08, 6, 0, 0)),
+                //TODO remove limit for observer Id
+                new SqlParameter("@IdObserverLimit", 10),
+
+            };
+
+            if (request.ApplyFilters)
+            {
+                if (request.To.HasValue)
+                {
+                    query += " AND a.LastModified <= @to ";
+                    parameters.Add(new SqlParameter("@to", request.To ?? DateTime.Now.AddDays(2)));
+                }
+
+                if (request.ObserverId.HasValue)
+                {
+                    query += " AND obs.Id = @ObserverId ";
+                    parameters.Add(new SqlParameter("@ObserverId", request.ObserverId));
+                }
+
+                if (request.NgoId.HasValue)
+                {
+                    query += " AND obs.IdNgo = @IdNgo ";
```

Comment:
We usually filter based on comparison. Are you sure you want to add exact matching here?